### ISSUE TITLE
[BUG] When hosts are stuck in maintenance mode and the cluster is unstable you can't access the UI

### ISSUE THREAD

**Issue by noahgildersleeve** [harvester/harvester#1493]
**Describe the bug**

There is an error state that causes the UI to be down while hosts are in maintenance mode and they won't come back up
**To Reproduce**
1. Cordon each node
2. turn on maintenance mode for each node
3. Power off each node
4. Turn them back on
**Expected behavior**

There shouldn't be any circumstances
**Support bundle**

[supportbundle_fd4d0150-71e6-4fa8-b965-2023248e7b83_2021-10-28T23-45-35Z.zip](https://github.com/harvester/harvester/files/7437932/supportbundle_fd4d0150-71e6-4fa8-b965-2023248e7b83_2021-10-28T23-45-35Z.zip)

**Environment:**
 - Harvester ISO version: master-c4a76706-head
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Dell powerEdge R620 - 200GB RAM 32Cores, 190GB RAM

**Additional context**
I was working with someone and @shangma and they were having this issue. The fix ended up being modifying the annotations on the hosts so that they weren't in maintenance mode anymore.  This was causing odd errors because the taints were making the nodes unschedulable which didn't allow them to come back up. 

I was unable to repro this issue with the steps, but added them in.

The VMs were turned off, each host was cordoned, then put into maintenance mode through the Harvester UI, then they were restarted. This all happened before the issues started and the Harvester cluster worked fine before.

This was on a three node cluster that had 3 RKE2 clusters deployed to it with 
- CentOS 8.4
- SLES
- Ubuntu Focal 20.04
```
kubectl taint nodes --all kubevirt.io/drain=scheduling:NoSchedule-
kubectl annotate node/r620-3 harvesterhci.io/maintain-status=completed
kubectl annotate node/r620-3 harvesterhci.io/maintain-status-
```
![Greenshot 2021-10-28 11 06 14](https://user-images.githubusercontent.com/83787952/139351657-595e7e25-ff3f-4d03-9ed5-51571b40cb3b.png)
![Greenshot 2021-10-28 10 47 23](https://user-images.githubusercontent.com/83787952/139351706-c134b722-3b14-4b6d-8ddf-f735659970d7.png)
![Greenshot 2021-10-28 10 56 22](https://user-images.githubusercontent.com/83787952/139351709-899defa3-dd11-4590-93b5-ec73fb5e2f2d.png)

**Comment by FrankYang0529** [comment]
With the update in this https://github.com/harvester/harvester/pull/1490, I can't reproduce this issue.

Steps:
* Create a 3-node cluster on KVM.
* Cordon and enable maintenance mode on each node, but we can't cordon/enable maintenance mode on the last available node. So the cluster has one schedulable node and two unschedulable nodes.
* Power off each node.
* Powen on each node.
* The cluster works fine and I can create a new VM running on it.

**Comment by TachunLin** [comment]
Verified **fixed** on master-119c2c48-head.This issue did not happen on the new cordon, maintenance mode design. 
Suggest to close this issue if no other concern.

### Result
Cordon node and enter maintenance mode, after machine reboot, we can login harvester dashboard. 
Node remain it's original status, we can disable/uncordon node, it can back to original status
![image](https://user-images.githubusercontent.com/29251855/141111698-64d9d648-9018-4c14-8828-539f6e44361e.png)

### Test Information
* Test Environment: 3 nodes harvester on provo bare machine
* Harvester version: master-119c2c48-head

### Verify Steps

1. Create 3 virtual machine on 3 nodes
2. Cordon 1st and 2nd node, 
![image](https://user-images.githubusercontent.com/29251855/141106858-cdfb35f3-50af-48d0-b776-1f1cc5dfcedc.png)
3. Enable maintenance mode on 1st and 2nd node 
![image](https://user-images.githubusercontent.com/29251855/141106968-e4d7a6be-6c60-4771-aabd-8df0ccafe252.png)
4. We can't cordon and enable maintenance node on the remaining node 
![image](https://user-images.githubusercontent.com/29251855/141107044-774166b8-117e-4635-b8a2-eeedb65e48fc.png)
5. Reboot 1st and 2nd node bare machine
6. Wait for harvester machine back to service
7. Login dashboard
8. Disable maintenance mode on 1st and 2nd node

### Additional Context
None

**Comment by TachunLin** [comment]
Close this issue since with the new design at least one node will keep in active that may somehow prevent it to happen.
